### DISCUSSION TITLE
mongodb in adonis 5

### DISCUSSION THREAD

**Discussion by vetsinen** [adonisjs/core#3024] [Help] [answered]
hello, can i use mongodb with adonis? google proposes lucid-mongo project, but this project seem to be expired, and useless with adonis 5

**Comment by McSneaky** [comment, marked as answer]
It's really easy. Just `npm install mongodb` :) 

Like Virk told:
> At the end of the day. AdonisJS is a Node framework and you can use any npm package with it :slightly_smiling_face: 

Since it's asked so frequently I wrote a small guide about it: 
https://mcsneaky.ap3k.pro/posts/adonis-v5-provider/

Seems like someone has made `adonis-mongodb` package too: 
https://github.com/zakodium/adonis-mongodb

Maybe there are some other packages too

MongoDB doesn't fit well with Lucid, since well.. MongoDB is meant to be used as document storage, Lucid is meant to work with relational data. Now you _can_ do all kinds of joins and stuff in MongoDB and force it to act as relational DB, but then why not use relational DB in fist place?

MongoDB could work wonders side-by-side with Lucid tho. Keep all your relational data in Postgres and use Lucid models, while you can use MongoDB for object storage and access that without Lucid

**Comment by arthurcvm** [comment]
Hi @vetsinen, you can use the `mongoose`, the mongoose provides models and validators to your application.
